src/nvim/option.c

```diff
@@ -454,6 +454,9 @@ void set_init_1(void)
   /* Set all options to their Vim default */
   set_options_default(OPT_FREE);
 
+  // set 'laststatus'
+  last_status(false);
```

Comment:
Yes, `set_num_option` seems to handle it, so I wonder why that isn't happening from `set_options_default`.